test/unit/options.spec.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+describe('options', function () {
+  describe('#defaultOptionsPath', function () {
+    var options = require('../../bin/options');
+
+    it("property should have value 'test/mocha.opts'", function () {
+      expect(options.defaultOptionsPath).to.equal('test/mocha.opts');
+    });
+
+    it('property should NOT be writable/configurable', function () {
+      expect(function () {
+        options.defaultOptionsPath = '/new/path/mocha.opts';
+      }).to.throwError(function (error) {
+        expect(error).to.be.a(TypeError);
+      });
+    });
+  });
+
+  describe('#path', function () {
+    var options = require('../../bin/options');
+
+    it('should return user passed options path if it exists, or the default path', function () {
+      var sinon = require('sinon');
+      var argv, path;
+
+      argv = [];
+      sinon.spy(argv, 'indexOf');
+      path = options.path(argv, options.defaultOptionsPath);
+      expect(argv.indexOf.calledOnce).to.be.true;
+      expect(path).to.equal(options.defaultOptionsPath);
+
+      argv = ['--opts', '/user/passed/mocha.opts'];
+      sinon.spy(argv, 'indexOf');
+      path = options.path(argv, options.defaultOptionsPath);
+      expect(argv.indexOf.calledTwice).to.be.true;
+      expect(path).to.equal('/user/passed/mocha.opts');
+    });
+  });
+
+  describe('#read', function () {
+    var options = require('../../bin/options');
+
+    it('throws when given a path that does not exist', function () {
+      // 1. no path is passed
+      expect(function () {
+        options.read();
+      }).to.throwError();
+
+      // 2. non-existing path is passed
+      expect(function () {
+        options.read('/path/does/not/exist/mocha.opts');
+      }).to.throwError();
+    });
+
+    it.skip('reads the given path when it exists', function () {
+      var fs = require('fs');
+      var del = require('del');
+
+      var now = Date.now();
+      var text = '--opt-1 value-1\n--opt-2 value-2';
+
+      fs.mkdirSync('test-data');
+      fs.writeFileSync('test-data/mocha' + now + '.opts', text);
+      expect(function () {
```

Comment:
I don't understand the need for the nested assertion?

Reply:
Sure we can do without it, but it does help with organising the report. Doesn't the following output make good sense?

![image](https://user-images.githubusercontent.com/3987523/33810982-afb39074-de0c-11e7-8cb0-bbf634f8da62.png)

Reply:
yes, but there's no need to wrap an assert in *another* assert:

```js
// this:
expect(() => expect(someAssertion)).not.to.throwError();
// should just be:
expect(someAssertion);
```
I *think* what you may be getting at here is something like `try` / `finally` to cleanup the `test-data` dir, which can be accomplished via `afterEach()` -- use another suite if necessary.

Reply:
I see what you mean now. Didn't notice when I wrote it originally and didn't notice yesterday neither.